### PR TITLE
fix: modified startup script to accommodate for the helm config

### LINKED ISSUE #1659: [TASK] Update Zeebe entrypoint to use config file instead of env vars
**Related to:**

PR: https://github.com/camunda/camunda-platform-helm/pull/1597

**Overview:**

Currently, the Zeebe entry point script overrides the network config and we need to use the env vars which is a snowflake compared to the other config.

**Actions:**

```[tasklist]
### Tasks
- [ ] https://github.com/camunda/camunda-platform-helm/pull/1910
```

### LINKED ISSUE #19114: startup.sh script overrides helm config
**Description**

I am trying to set the following in the helm [configmap](https://github.com/camunda/camunda-platform-helm/pull/1910) for the zeebeGateway:
```
 zeebe:
      gateway:
          host: 0.0.0.0
```

The [startup.sh](https://github.com/camunda/camunda/blob/main/zeebe/docker/utils/startup.sh) script overrides this config with the use of env vars:
```
HOST=$(hostname -i)

if [ "$ZEEBE_STANDALONE_GATEWAY" = "true" ]; then
    export ZEEBE_GATEWAY_NETWORK_HOST=${ZEEBE_GATEWAY_NETWORK_HOST:-${HOST}}
    export ZEEBE_GATEWAY_CLUSTER_HOST=${ZEEBE_GATEWAY_CLUSTER_HOST:-${ZEEBE_GATEWAY_NETWORK_HOST}}

    exec /usr/local/zeebe/bin/gateway

```

We could remove these env vars in the startup.sh script and let the helm configmap have control. I am aware that this might cause potential problems running zeebe outside of helm.

### PR DESCRIPTION
**Pull request by hamza-m-masood**
## Description
I added a check for an existing env var called `ZEEBE_SET_SEED_ENV_VARS` which exists in this [PR](https://github.com/camunda/camunda-platform-helm/pull/1910)
If `ZEEBE_SET_SEED_ENV_VARS` is set to false then the env var `ZEEBE_GATEWAY_NETWORK_HOST` will not be set. This will result in the startup script not overriding our helm config.
## Related issues
https://github.com/camunda/camunda/issues/19114
closes #19114

### CODE REVIEW

File: zeebe/docker/utils/startup.sh [hunk 1]
```diff
@@ -1,9 +1,12 @@
 #!/bin/sh -xeu
 
 HOST=$(hostname -i)
+ZEEBE_SET_SEED_ENV_VARS=${ZEEBE_SET_SEED_ENV_VARS:-true}
```
Comment (aabouzaid) [inline zeebe/docker/utils/startup.sh:4]: Let's call it `ZEEBE_DOCKER_SET_SEED_ENV_VARS` so it's clear it's used for Docker.

File: zeebe/docker/utils/startup.sh [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 
 if [ "$ZEEBE_STANDALONE_GATEWAY" = "true" ]; then
-    export ZEEBE_GATEWAY_NETWORK_HOST=${ZEEBE_GATEWAY_NETWORK_HOST:-${HOST}}
+    if [ "$ZEEBE_SET_SEED_ENV_VARS" = "true" ]; then
+        export ZEEBE_GATEWAY_NETWORK_HOST=${ZEEBE_GATEWAY_NETWORK_HOST:-${HOST}}
+    fi
     export ZEEBE_GATEWAY_CLUSTER_HOST=${ZEEBE_GATEWAY_CLUSTER_HOST:-${ZEEBE_GATEWAY_NETWORK_HOST}}
```
Comment (aabouzaid) [inline zeebe/docker/utils/startup.sh:10]: Probably we need it for the env var `ZEEBE_GATEWAY_CLUSTER_HOST` too.

File: zeebe/docker/utils/startup.sh [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 
     exec /usr/local/zeebe/bin/gateway
```
Comment (aabouzaid) [inline zeebe/docker/utils/startup.sh:12]: The same should be done for `ZEEBE_BROKER_*` vars.

**Review by npepinpe — APPROVED** [review]
Keep in mind we also set some environment variables directly in the Dockerfile itself =/ Could this be an issue for other settings you wish to configure?

Otherwise I think this is an acceptable interim approach. Ultimately we should not set any environment variable in the Dockerfile itself, I think.

Other than that, please review Ahmed's comments before merging.

**Comment by hamza-m-masood** [comment]
@npepinpe Could you please elaborate on what env vars are set in the Dockerfile?

File: zeebe/docker/utils/startup.sh (hunk 1 above)
Comment (hamza-m-masood) [reply zeebe/docker/utils/startup.sh]: ✅

File: zeebe/docker/utils/startup.sh (hunk 2 above)
Comment (hamza-m-masood) [reply zeebe/docker/utils/startup.sh:10]: On a working instance, when I echo the value of `ZEEBE_GATEWAY_CLUSTER_HOST`, I see that it is the same as the output of `hostname -i`
So I don't think it has to be `0.0.0.0` on a helm installation.

File: zeebe/docker/utils/startup.sh (hunk 3 above)
Comment (hamza-m-masood) [reply zeebe/docker/utils/startup.sh:12]: Only `ZEEBE_GATEWAY_NETWORK_HOST` needs to change to `0.0.0.0`. I don't understand why we need to add other env vars into the if statement as well.

**Comment by npepinpe** [comment]
Hi both, we will likely go with this solution here: https://github.com/camunda/camunda/pull/19228

I think as it removes the variables from the Dockerfile and the startup.sh, it should solve the issue at hand, right?

**Comment by npepinpe** [comment]
@hamza-m-masood - we aren't planning on back porting this, but is it something you would need? Or are the changes you wanted to make only for the next version and up?

**Comment by hamza-m-masood** [comment]
@npepinpe 
This fix would need to be added to version 8.5 and onwards, not just the next version.

**Comment by npepinpe** [comment]
So it needs to be backported to 8.5? Considering this is not without risk, I just want to confirm that we're sure this is really necessary :)

**Comment by npepinpe** [comment]
Typically we backport bug fixes and security patches by default, and everything else is decided on a case by case basis, where we would evaluate the risk/reward ratio (since every backport carries _some_ risk). Since this is less a bug and more of a QoL improvement, I would not automatically backport it, unless it brings a lot of value (whether for users, or for you, e.g. greatly simplifying your life).

**Comment by aabouzaid** [comment]
@npepinpe Shouldn't we still fix it for the next cycle (8.6)? :thinking:

**Comment by npepinpe** [comment]
It's fixed for the next release via https://github.com/camunda/camunda/pull/19228

**Comment by aabouzaid** [comment]
@npepinpe thanks for the clarification :raised_hands: (I missed the link above)

**Pull request closed without merging**